server/src/main/java/org/cloudfoundry/identity/uaa/account/PasswordChangeEndpoint.java

```diff
@@ -87,11 +96,8 @@ public ActionResult changePassword(@PathVariable String userId, @RequestBody Pas
         String newPassword = change.getPassword();
 
         throwIfPasswordChangeNotPermitted(userId, oldPassword, zoneId);
-        if (dao.checkPasswordMatches(userId, newPassword, zoneId)) {
-            throw new InvalidPasswordException("Your new password cannot be the same as the old password.", UNPROCESSABLE_ENTITY);
-        }
```

Comment:
This is the old behavior, of course.  Now added to `changePasswordWithHistoryCheck()`.  Figured the history check should not be executed until the original password is authenticated as to not leak comprising error messages unnecessarily.